### PR TITLE
Bar us 12

### PR DESCRIPTION
**Pull request by BRohrig**
feature tests and functionality for user story 12 complete

### CODE REVIEW

File: app/models/merchant.rb [hunk 1]
```diff
@@ -17,4 +17,13 @@ def toggle_status
   def self.group_by_status(status)
     self.where(status: status)
   end
+
+  def top_5_revenue_items
```
Comment (Sergio-Azcona) [inline app/models/merchant.rb:34]: nice, clean job on this method!!

File: app/views/merchants_items/index.html.erb [hunk 2]
```diff
@@ -30,5 +30,17 @@
           <% end %>
       </div>
     </div>
+
+    <div class="items-right">
+      <div id="top_5_items">
+        <h3>Top 5 Items by Revenue:</h3><br>
+          <% @top_5_items.each.with_index(1) do |item, i| %>
```
Comment (Sergio-Azcona) [inline app/views/merchants_items/index.html.erb:37]: nice using/finding the **with_index** method here!

Comment (Sergio-Azcona) [inline app/views/merchants_items/index.html.erb:37]: Question: **with_index** is pulling the 1st index position (i.e. the second item in the array).

did you want the second item or the first item in the array (index position 0)?

**Review by Sergio-Azcona — COMMENTED** [review]
Question about what show page> index position being displayed

**Review by Sergio-Azcona — APPROVED** [review]

**Comment by Sergio-Azcona** [comment]
.each_with(#) - Brady used in the show page to number the output displayed:
[difference between each.with_index and each_with_index in Ruby?]
https://stackoverflow.com/questions/20258086/difference-between-each-with-index-and-each-with-index-in-ruby

[each_with_index_do starting at 1 for index]
(https://stackoverflow.com/questions/14387589/each-with-index-do-starting-at-1-for-index)

File: app/models/invoice_item.rb [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
   belongs_to :invoice
+  has_many :transactions, through: :invoices
   has_one :merchant, through: :item
   has_one :customer, through: :invoice
 
   enum status: {packaged: 0,
                 pending: 1,
                 shipped: 2}
+
+  def revenue
```
Comment (bflanagan138) [inline app/models/invoice_item.rb:12]: handy method!

**Review by bflanagan138 — APPROVED** [review]
Looks good

**Pull request merged**